### ISSUE TITLE
devices/generic.c warning: ISO C90 forbids mixed declarations and code

### ISSUE THREAD

**Issue by koppi** [synapticon/Etherlab_EtherCAT_Master#1]
Please note: [this code](https://github.com/synapticon/Etherlab_EtherCAT_Master/blob/735536e48bd01688bbc6a29ca61ec1328ca6004d/devices/generic.c#L218) produces the following warning at compile time:
```
devices/generic.c: In function ‘ec_gen_device_create_socket’:
devices/generic.c:218:5: warning: ISO C90 forbids mixed declarations and code [-Wdeclaratio
n-after-statement]
     struct net *nd_net = dev_net(dev->netdev);
     ^
```

**Comment by koppi** [comment]
Please add an extra declaration statement:
```c
#if (LINUX_VERSION_CODE >= KERNEL_VERSION(4, 2, 0)) 
    struct net *nd_net;
#endif
```

**Comment by fjes** [comment]
Thanks for the feedback, you are right about these issues and we will fix them asap.

**Comment by fjes** [comment]
Fixed with 5c588dc2142a49c23a9d80786804b2c5cf16c928
